infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/convert/DockerImageEnvironmentConverter.java

```diff
@@ -64,9 +65,10 @@ public KubernetesEnvironment convert(DockerImageEnvironment environment)
             .build();
     return KubernetesEnvironment.builder()
         .setMachines(environment.getMachines())
-        .setInternalRecipe(environment.getRecipe())
+        .setInternalRecipe(new InternalRecipe(KubernetesEnvironment.TYPE, null, ""))
```

Comment:
It may be surprised that there is an empty recipe for environment. As far as I understand you need to check in component what type does environment have, and with current implementation it works but what if there would be some component that would need access to the original recipe?
Maybe it makes more sense to add `getType()` method to `InternalEnvironment`? Then it will show a real environment type, and `getRecipe().getType()` would show original recipe type. WDYT?

Reply:
I like the idea!